package.json

```diff
@@ -28,7 +28,7 @@
     "./CHANGELOG.md"
   ],
   "scripts": {
-    "build": "npm run build:clean && npm run generate && vite build",
+    "build": "npm run build:clean && npm run generate && npm run tsc && vite build",
```

Comment:
This felt like the best place to insert the typecheck. Broken types won't break your local dev build while you're working but will break both CI and publish here. There's some room for efficiency to run tsc and vite in parallel here but didn't feel worth right now.